File: backend/demo/src/main/java/com/example/demo/model/Notification.java
package com.example.demo.model;



import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.index.Indexed;

import java.time.Instant;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "notifications")
public class Notification {
    @Id
    private String id;
    private String userId;
    private String actorId;
    private String actorName;
    private String type;  // LIKE, COMMENT, FOLLOW, COMMUNITY_JOIN, POST_SHARE
    private String referenceId;  // postId, commentId, etc.
    private String content;
    private boolean read;
    private LocalDateTime createdAt;
}

File: frontend/src/types/Notification.js
/**
 * @typedef {"LIKE" | "COMMENT"} NotificationType
 */

/**
 * @typedef {Object} Notification
 * @property {string} id
 * @property {string} userId
 * @property {string} message
 * @property {NotificationType} type
 * @property {boolean} read
 * @property {string} [link]
 * @property {string} createdAt
 */

/**
 * @typedef {Object} NotificationResponse
 * @property {Notification[]} notifications
 * @property {boolean} hasMore
 * @property {number} totalCount
 */
